### PR TITLE
CRM-16505: Fix 'other amount' translation for multilingual/non-english contribution forms.

### PR DESCRIPTION
**Pull request by mlutfy**

### CODE REVIEW

File: CRM/Contribute/Form/ContributionPage/Amount.php [hunk 1]
```diff
@@ -705,7 +705,7 @@ public function postProcess() {
               }
               else {
                 CRM_Core_DAO::setFieldValue('CRM_Price_DAO_PriceField', $priceFieldID, 'is_required', 0);
-                CRM_Core_DAO::setFieldValue('CRM_Price_DAO_PriceFieldValue', $priceFieldValueID, 'label', 'Other Amount');
+                CRM_Core_DAO::setFieldValue('CRM_Price_DAO_PriceFieldValue', $priceFieldValueID, 'label', ts('Other Amount'));
```
Comment (kurund) [inline CRM/Contribute/Form/ContributionPage/Amount.php]: I don't think we should update translated value in db.

Comment (mlutfy) [inline CRM/Contribute/Form/ContributionPage/Amount.php]: @kurund I'm not sure I understand. Labels are always translated in the DB. For multi-lingual, we have label_fr_CA, label_en_US, etc.

Comment (kurund) [inline CRM/Contribute/Form/ContributionPage/Amount.php]: @mlutfy in DB value is saved as 'Other Amount' so I think ts('Other Amount') should be changed to 'Other Amount'

Comment (mlutfy) [inline CRM/Contribute/Form/ContributionPage/Amount.php]: @kurund: I did a few more tests, and without those ts() calls, the Other Amount field option does not get translated.

**Pull request merged**